### PR TITLE
Add `join_path_var` function to join list of paths into a colon-separated string value

### PR DESCRIPTION
**Pull request by Flamefire**
`os.pathsep.join` is often used but there might be empty entries.
Those should usually be omitted as they can have different semantics
than expected. E.g. `$PYTHONPATH` treats those as `$PWD`.

I ran into this because of a build failure in TensorFlow and though this method should best be in framework

### CODE REVIEW

File: test/framework/environment.py [hunk 1]
```diff
@@ -31,13 +31,24 @@
...
+from pathlib import Path
 
 import easybuild.tools.environment as env
 
 
 class EnvironmentTest(EnhancedTestCase):
     """ Testcase for run module """
 
+    def test_join_path_var(self):
+        """Test join_path_var function."""
```
Comment (boegel) [inline test/framework/environment.py:43]: I'd add one more trivial test case: when no empty paths are involved, `join_path_var` provides same result as `os.pathsep.join`:

```suggestion
        """Test join_path_var function."""
        # join_path_var is same as os.pathsep.join if there are no empty paths involved
        paths = ['/foo', '/bar', '/baz']
        self.assertEqual(env.join_path_var(paths), os.pathsep.join(paths))

```

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm
